### PR TITLE
Add git history analysis to code review pipeline

### PR DESCRIPTION
**Pull request by haacked**
## Summary
- Add per-file git churn metrics (commit count, author count, high-churn flag) to the review orchestrator, computed via new `git-file-history.sh` script
- Context explorer now gathers targeted `git log` for high-churn files and `git log -S` for surprising code, classifying churn patterns (stability risk, coordination risk, neutral)
- Remove unused `--overwrite` and `--append` flags from review argument parsing

## How it works
- `git-file-history.sh` reads file paths from stdin, runs `git log` per file (author email + date query), outputs JSON with `recent_commits`, `recent_authors`, `last_modified`, and `high_churn` (true when 10+ commits or 3+ authors in 30 days)
- Results are merged into `file_metadata` so all review agents see churn signals without additional git calls
- Context explorer uses the `high_churn` flags to decide where to run `git log --oneline -5` and classifies patterns for downstream agents
- Authors are counted by email (`%ae`) for reliable deduplication across name variants
- The 50-file processing cap uses `continue` (not `break`) to drain stdin and avoid SIGPIPE under `set -o pipefail`

## Test plan
- 12 new unit tests in `test-git-file-history.bats` covering JSON output, commit/author counting, churn thresholds, edge cases (nonexistent files, empty input), and 50-file cap
- Tests use distinct `user.email` values to match the script's email-based author counting
- All tests pass

### CODE REVIEW

File: skills/review-code/scripts/review-orchestrator.sh [hunk 1]
```diff
@@ -274,6 +274,28 @@ build_review_data() {
     local file_metadata
     file_metadata=$(echo "${diff_content}" | "${SCRIPT_DIR}/pre-review-context.sh")
 
+    # Compute git history metrics for modified files (only when inside a git repo)
+    local git_history='{}'
+    if git rev-parse --is-inside-work-tree > /dev/null 2>&1; then
+        git_history=$(echo "${file_metadata}" | jq -r '.modified_files[].path' | "${SCRIPT_DIR}/git-file-history.sh")
```
[the remark this answers is not in the document]
Comment (haacked) [reply skills/review-code/scripts/review-orchestrator.sh:280]: Not needed -- the root cause is the `break` in `git-file-history.sh` which closes the pipe early. Fixed that to `continue` so it drains remaining stdin without processing, avoiding SIGPIPE regardless of how many files the caller sends.

File: skills/review-code/scripts/git-file-history.sh [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        if [[ -n "${log_output}" ]]; then
+            recent_commits=$(echo "${log_output}" | wc -l | tr -d ' ')
+            recent_authors=$(echo "${log_output}" | cut -f1 | sort -u | wc -l | tr -d ' ')
+            last_modified=$(echo "${log_output}" | head -1 | cut -f2)
+        fi
+
+        # For files dormant >30 days, fetch absolute last_modified
+        if [[ -z "${last_modified}" ]]; then
+            last_modified=$(git log -1 --format="%as" -- "${file_path}" 2> /dev/null || echo "")
+        fi
```
[the remark this answers is not in the document]
Comment (haacked) [reply skills/review-code/scripts/git-file-history.sh:70]: The PR description is ephemeral context, not documentation. The code comment on line 67 already explains why the fallback exists. Not worth a code change.

File: skills/review-code/scripts/review-orchestrator.sh [hunk 3]
```diff
@@ -274,6 +274,28 @@ build_review_data() {
...
+        debug_trace "build_review_data" "Not inside a git repository; skipping git history analysis"
+    fi
+
+    # Merge git history into file_metadata
+    file_metadata=$(echo "${file_metadata}" | jq --argjson history "${git_history}" '
+        .modified_files |= map(
+            . + {git_history: ($history[.path] // {recent_commits: 0, recent_authors: 0, last_modified: null, high_churn: false})}
+        )
+        | . + {has_high_churn_files: ([.modified_files[] | select(.git_history.high_churn == true)] | length > 0)}
+    ')
```
[the remark this answers is not in the document]
Comment (haacked) [reply skills/review-code/scripts/review-orchestrator.sh:297]: PRs with >50 modified files are rare, and the 50-file cap is already generous. Adding truncation metadata is over-engineering for a corner case that does not meaningfully affect review quality.

File: agents/code-review-context-explorer.md [hunk 4]
```diff
@@ -38,6 +38,16 @@ Gather only when relevant to the changes:
 - **Security-Sensitive**: Find existing security patterns when auth or validation code changes
 - **Performance-Critical**: Find similar optimizations when queries or loops are modified
 
+### 5. Git History Context
+
+Check `file_metadata` for files with `git_history.high_churn: true`. For each high-churn file:
+- Run `git log --oneline -5 <file>` to surface recent changes
```
[the remark this answers is not in the document]
Comment (haacked) [reply agents/code-review-context-explorer.md:44]: This is prose guidance for an AI agent, not executable shell code. The agent handles path quoting on its own when it generates commands.

File: skills/review-code/handlers/review.md [hunk 5]
```diff
@@ -166,6 +166,8 @@ Explore the codebase to understand:
   (grep for function/method names, report top 3-5 callers per significantly modified function)
 - Existing patterns for similar functionality
 - Reusable utilities or conventions
+- Git history for high-churn files and surprising code
+  (check `git_history.high_churn` flags in file_metadata; run `git log` for flagged files and for any code whose purpose is non-obvious)
```
[the remark this answers is not in the document]
Comment (haacked) [reply skills/review-code/handlers/review.md:170]: Same as above. This is guidance for an AI agent that generates its own shell commands with proper quoting.
